Contents/mods/AwayFromZomboid/media/lua/client/AwayFromZomboid/AwayFromZomboid.lua

```diff
@@ -353,4 +358,14 @@ end
 
 -- Init hook
 
-Events.OnConnected.Add(AwayFromZomboid.init)
\ No newline at end of file
+Events.OnCreatePlayer.Add(AwayFromZomboid.init)
+Events.OnPlayerDeath.Add(function (player)
+    if getPlayer():isDead() then
```

Comment:
This can probably be removed, and the OnCreatePlayer hook can be unhooked at the end of the init function. For code style cohesion, I'd also prefer not to have function definitions inside of event hooking.